utility.go

```diff
@@ -31,19 +30,17 @@ import (
 // verifyAccess verifies the caller. It verifies both that the caller is authenticated by the
 // Identity Provider, and that the caller has the necessary scopes.
 func (d *D1) verifyAccess(ctx context.Context, token string, scope id.Scope) (id.Identity, error) {
-	l := zerolog.Ctx(ctx)
-
-	l.Debug().Msg("authenticating caller")
+	log.Ctx(ctx).Debug().Msg("authenticating caller")
 	identity, err := d.idProvider.GetIdentity(ctx, token)
 	if err != nil {
-		l.Debug().Err(err).Msg("authentication failed")
+		log.Ctx(ctx).Debug().Err(err).Msg("authentication failed")
 		return id.Identity{}, ErrNotAuthenticated
 	}
-	log.WithUID(l, identity.ID)
+	log.WithUserID(ctx, identity.ID)
 
-	l.Debug().Msg("authorizing caller")
+	log.Ctx(ctx).Debug().Msg("authorizing caller")
```

Comment:
Do you have any idea if it reduces performance getting the logger from the context multiple times in a function, or if it is to store the logger in a variable? It might be a negligible impact

Reply:
It's probably a small performance hit, but it essentially just returns a pointer from a map: https://github.com/rs/zerolog/blob/d894f123bc5c2a887c95e90218b9410563141d67/ctx.go#L44

Reply:
Doesn't seem to be anything significant:
```
name       old time/op  new time/op  delta
Encrypt-8   566µs ±21%   596µs ±26%   ~     (p=0.056 n=50+50)
Decrypt-8   593µs ±25%   571µs ±14%   ~     (p=0.071 n=49+49)
```

Reply:
:+1: